app/src/main/java/com/futurumgame/base/serialization/parsing/IntegerUnitsMapEntryParseRule.java

```diff
@@ -0,0 +1,54 @@
+package com.futurumgame.base.serialization.parsing;
+
+import com.futurumgame.base.additionalDatatypes.Units;
+import com.futurumgame.base.collections.PseudoMapEntry;
+import com.futurumgame.base.enums.Separator;
+import com.futurumgame.base.exceptions.UnitsFormatException;
+import com.futurumgame.base.util.Logger;
+import com.futurumgame.base.util.StringUtil;
+
+import java.util.Map;
+
+public class IntegerUnitsMapEntryParseRule extends MapEntryParseRule<Integer, Units> {
+
+    public IntegerUnitsMapEntryParseRule() {
+    }
+
+    @Override
+    public ParseResult<Map.Entry<Integer, Units>> next(String string) {
+        super.next(string);
+        if (getReadChars().isEmpty()) {
+            return ParseResult.failResult();
+        }
+        String[] data = getReadChars().split(Separator.DataStructureSeparator.getSeparator());
+        if (data.length < 2) {
```

Comment:
Again... What is this magical value of 2?

Reply:
same here